packages/fuels-accounts/src/wallet.rs

```diff
@@ -263,19 +264,32 @@ impl Account for WalletUnlocked {
 #[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
 #[cfg_attr(not(target_arch = "wasm32"), async_trait)]
 impl Signer for WalletUnlocked {
-    type Error = WalletError;
-    async fn sign_message<S: Send + Sync + AsRef<[u8]>>(
-        &self,
-        message: S,
-    ) -> WalletResult<Signature> {
+    type Error = Error;
+    async fn sign_message<S: Send + Sync + AsRef<[u8]>>(&self, message: S) -> Result<Signature> {
         let message = Message::new(message);
         let sig = Signature::sign(&self.private_key, &message);
+
         Ok(sig)
     }
 
     fn sign_transaction(&self, tb: &mut impl TransactionBuilder) {
         tb.add_unresolved_signature(self.address().clone(), self.private_key);
     }
+
+    fn sign_built_transaction(&self, tx: &mut impl Transaction) -> Result<Signature> {
```

Comment:
I know we already mentioned it in the docs, but I believe we should add a comment here, again, about calling this method in the right witness order; I can see this `tx.append_witness` (along with tx being `mut`) causing some issues due to ordering and being something hard to debug. Official docs + method-level docs should make things easier.